### DISCUSSION TITLE
Rich Text Editor

### DISCUSSION THREAD

**Discussion by colinmac17** [livewire/flux#220] [Feature Requests]
Similar to Filament, it would be great to implement a rich text editor for things like emails or blog posts. I am a customer of alpine components so I could certainly reach for that but this would be very helpful!

**Comment by ProgrammerZ** [comment]
Whatever Flux team would use for the Rich Editor, I would consider having **`Merge Tags`**(aka merge fields) functionality included as an option. Merge tags are essential in email/template context.

For example TinyMCE has good merge tags implementation, but it's paid. [Lexical](https://lexical.dev/) is pretty good and have a lot of out of the box. Here is a [mentions plugin](https://github.com/sodenn/lexical-beautiful-mentions)

**Comment by calebporzio** [comment]
Rich text editors are an almost unreasonably complex undertaking for us to tackle ourselves at this point.

We could integrate with something like TinyMCE or TipTap editor, however so far we have tried to keep all Flux's JS and styling really lean and hand-crafted.

I could see a future where we offer additional integrations as an optional thing that users can publish into their apps and configure/maintain themselves. This way we aren't on hook for the zillion requested configurations and bug-fixes people will inevitably request.

While we think about this from a product standpoint. If you came here because you want this feature, please reply with your needs so we can be aware of what people want.

**Reply by colinmac17** [reply]
I agree @calebporzio! I took the approach of building a Quill.js component in alpine and it hooks up beautifully with live wire and with styling it fits into flux just fine. Great work I'm loving everything about flux!

**Reply by oMikeo** [reply]
We're also using Quill.js (via customised version of https://github.com/joelwmale/livewire-quill) and it drops in pretty seamlessly with the snow theme.

**Reply by calebporzio** [reply]
Oh nice, I'll look into Quill - maybe we can offer like a DIY guide for using Quill or something with Flux

**Comment by calebporzio** [comment]
Closing this. Feel free to open new discussions for specific feature requests related to the editor, but the core editor has been added. Thanks everyone for weighing in on this,.
